File: rust/mandelbrot/Simple.rs
fn mandelbrot(x: f64, y: f64) -> i64 {
    let cr = y - 0.5;
    let ci = x;

    let mut zi = 0.0;
    let mut zr = 0.0;
    let mut i: i64 = 0;

    loop {
        i += 1;

        let temp = zr * zi;

        let zr2 = zr * zr;
        let zi2 = zi * zi;

        zr = zr2 - zi2 + cr;
        zi = temp + temp + ci;

        if zi2 + zr2 > 16.0 {
            return i;
        }

        if i > 5000 {
            return 0;
        }
    }
}

fn index() {
    for y in -39..39 {
        println!();

        for x in -39..39 {
            let i = mandelbrot(
                x as f64 / 40.0,
                y as f64 / 40.0
            );

            if i== 0 {
                print!("*");
            } else {
                print!(" ");
            }
        }
    }
}

fn main() {
    index();
}

File: ruby/mandelbrot/Simple.rb
def index
  for y in -39..39
    puts

    for x in -39..39
      i = mandelbrot(
        x / 40.0,
        y / 40.0
      )

      if i == 0
        print("*")
      else
        print(" ")
      end
    end
  end

  puts
end

def mandelbrot(x, y)
  cr = y - 0.5
  ci = x
  zi = 0.0
  zr = 0.0
  i = 0

  loop do
    i += 1

    temp = zr * zi

    zr2 = zr * zr
    zi2 = zi * zi

    zr = zr2 - zi2 + cr
    zi = temp + temp + ci

    if zi2 + zr2 > 16
      return i
    end

    if i > 5000
      return 0
    end
  end
end

startTimeMs = Time.now.to_f * 1000

index

endTimeMs = Time.now.to_f * 1000
durationMs = (endTimeMs - startTimeMs).to_i

puts "Execution time: #{durationMs}ms"